client/components/tinymce/plugins/embed-reversal/plugin.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * External dependencies
+ */
+import tinymce from 'tinymce/tinymce';
+import { forEach, includes, partial } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import PostEditStore from 'lib/posts/post-edit-store';
+import wpcom from 'lib/wp';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+function embedReversal( editor ) {
+	const store = editor.getParam( 'redux_store' );
+	if ( ! store ) {
+		return;
+	}
+
+	function replaceMarkup( markup, result ) {
+		if ( ! result || ! editor ) {
+			return;
+		}
+
+		const isVisualEditMode = ! editor.isHidden();
+
+		if ( isVisualEditMode ) {
+			// Check textContent of all elements in Visual editor body
+			forEach( editor.getBody().querySelectorAll( '*' ), ( element ) => {
+				if ( includes( element.textContent, markup ) ) {
+					element.textContent = element.textContent.replace( markup, result.result );
+					editor.undoManager.add();
+					return false;
```

Comment:
>It doesn't seem like this is necessary

In rebased 0115159120c610a0a68fb665e03425cf9dac8c26, I opted to replace `_.forEach` with a simple `for ... of` loop, using much more identifiable `break;` instead of the obscure `return false;`.